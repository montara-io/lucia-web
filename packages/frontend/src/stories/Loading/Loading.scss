@import '../../styles/colors';

.m-loading-light .p-progressbar,
.m-loading-dark .p-progressbar {
  width: 140px;
  height: 3px;
}

.m-loading-light {
  .p-progressbar {
    background: $gray240;

    .p-progressbar-value {
      background: $blue;
    }
  }
}

.m-loading-dark {
  .p-progressbar {
    background: $blue;

    .p-progressbar-value {
      background: $white;
    }
  }
}
