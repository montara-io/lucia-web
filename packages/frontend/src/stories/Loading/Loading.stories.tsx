import React from 'react'
import Loading from './Loading'

export default {
  title: 'Components/Loading',
  component: Loading,
}

export const Primary = (args: any) => <Loading {...args} />

Primary.args = {
  id: 'id',
  mode: 'dark',
}
Primary.parameters = {
  jest: ['Loading.test.tsx'],
}
