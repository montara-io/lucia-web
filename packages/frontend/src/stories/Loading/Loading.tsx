import React, { FunctionComponent } from 'react'
import './Loading.scss'
import { ProgressBar } from 'primereact/progressbar'
import { IntlProvider } from 'react-intl'
import lang from '../lang/en.json'
import classNames from 'classnames'

export type LoadingProps = {
  id: string
  mode: 'dark' | 'light'
  className?: string
}
const locale = navigator.language

const Loading: FunctionComponent<LoadingProps> = ({ mode, id, className }) => {
  return (
    <div
      id={id}
      data-testid="m-loading"
      className={classNames(
        `m-loading-${mode === 'dark' ? 'dark' : 'light'}`,
        className,
      )}
    >
      <ProgressBar mode="indeterminate" />
    </div>
  )
}
const WrappedLoading = (props) => {
  return (
    <IntlProvider locale={locale} messages={lang}>
      <Loading {...props} />
    </IntlProvider>
  )
}
export default WrappedLoading
