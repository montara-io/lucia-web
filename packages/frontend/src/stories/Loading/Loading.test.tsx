import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Loading from "./Loading";

describe("Test Input Component", () => {
  it("Snapshot renders correctly - dark mode", () => {
    const wrapper = render(
      <Loading
        {...{
          id: "Loading",
          mode: "dark",
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("Snapshot renders correctly - light mode", () => {
    const wrapper = render(
      <Loading
        {...{
          id: "Loading",
          mode: "light",
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
