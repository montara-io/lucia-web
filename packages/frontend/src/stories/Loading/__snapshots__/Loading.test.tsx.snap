// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Input Component Snapshot renders correctly - dark mode 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="m-loading-dark"
        data-testid="m-loading"
        id="Loading"
      >
        <div
          class="p-progressbar p-component p-progressbar-indeterminate"
          role="progressbar"
        >
          <div
            class="p-progressbar-indeterminate-container"
          >
            <div
              class="p-progressbar-value p-progressbar-value-animate"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="m-loading-dark"
      data-testid="m-loading"
      id="Loading"
    >
      <div
        class="p-progressbar p-component p-progressbar-indeterminate"
        role="progressbar"
      >
        <div
          class="p-progressbar-indeterminate-container"
        >
          <div
            class="p-progressbar-value p-progressbar-value-animate"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test Input Component Snapshot renders correctly - light mode 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="m-loading-light"
        data-testid="m-loading"
        id="Loading"
      >
        <div
          class="p-progressbar p-component p-progressbar-indeterminate"
          role="progressbar"
        >
          <div
            class="p-progressbar-indeterminate-container"
          >
            <div
              class="p-progressbar-value p-progressbar-value-animate"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="m-loading-light"
      data-testid="m-loading"
      id="Loading"
    >
      <div
        class="p-progressbar p-component p-progressbar-indeterminate"
        role="progressbar"
      >
        <div
          class="p-progressbar-indeterminate-container"
        >
          <div
            class="p-progressbar-value p-progressbar-value-animate"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
